### DISCUSSION TITLE
Affordance for focused state on filled input

### DISCUSSION THREAD

**Discussion by rob-pathway** [anday013/react-native-otp-entry#55] [Ideas]
@anday013 Original issue: https://github.com/anday013/react-native-otp-entry/issues/53

If we agree on the job/spec, I can submit a PR.

Job to be done: When I edit a filled OTP input, I want a signal that it's focused, so I know I can start editing

Spec:
- On OTP press, apply styles to final input field if OTP is filled
- Expose theme properties `lastPinCodeFilledContainerStyle` and `lastPinCodeFilledTextStyle` to style the input. This adds flexibility at the expense of complexity or redundancy.
- Default to no style so as not to disrupt previous versions and cause unexpected UX

**Comment by anday013** [comment]
@rob-pathway  Thank you for the detailed description. I suggest to keep it simple. No need to add a separate prop for styling last cell just for focus. Let's just apply focus styles (focusColor/focusedPinCodeContainerStyle) on the last cell without showing the stick (caret). wdyt?

**Reply by rob-pathway** [reply]
Definitely simpler that way.

Should it be opt-in?

**Reply by anday013** [reply]
@rob-pathway nah, let's make it by default. It's a nice feature, and I can't think of a negative outcome for the current users. We just need to properly test and check all the edge cases accordingly, in order to make it bug-free :)

**Reply by rob-pathway** [reply]
Sounds good! Will try to get to this tonight.

**Comment by rob-pathway** [comment]
@anday013 Sorry, just getting to this.

Do you have a recommended basic dev setup that embeds the repo inside a bare react-native/expo project? For testing the UI.

**Reply by anday013** [reply]
Hey @rob-pathway , sorry I am out these days. I'll look at your PR as soon as I can. Thank you.

Regarding the dev environment, the easiest way is to initialize an Expo project and clone the repo into it :)
